aptos-move/framework/aptos-framework/sources/state_storage.move

```diff
@@ -23,59 +21,70 @@ module aptos_framework::state_storage {
-    /// epoch older than ideal, but the Vm doesn't need to worry about reloading
-    /// gas parameters after the first txn of an epoch.
-    struct GasParameter has key, store {
-        usage: Usage,
-    }
-
     public(friend) fun initialize(aptos_framework: &signer) {
         system_addresses::assert_aptos_framework(aptos_framework);
         assert!(
             !exists<StateStorageUsage>(@aptos_framework),
             error::already_exists(ESTATE_STORAGE_USAGE)
         );
-        assert!(
-            !exists<GasParameter>(@aptos_framework),
-            error::already_exists(EGAS_PARAMETER)
-        );
         move_to(aptos_framework, StateStorageUsage {
             epoch: 0,
             usage: Usage {
                 items: 0,
                 bytes: 0,
             }
         });
-        move_to(aptos_framework, GasParameter {
-            usage: Usage {
-                items: 0,
-                bytes: 0,
-            }
-        });
     }
 
     public(friend) fun on_new_block(epoch: u64) acquires StateStorageUsage {
+        assert!(
+            exists<StateStorageUsage>(@aptos_framework),
+            error::not_found(ESTATE_STORAGE_USAGE)
+        );
         let usage = borrow_global_mut<StateStorageUsage>(@aptos_framework);
         if (epoch != usage.epoch) {
             usage.epoch = epoch;
             usage.usage = get_state_storage_usage_only_at_epoch_beginning();
         }
     }
 
-    public(friend) fun on_reconfig() acquires StateStorageUsage, GasParameter {
-        let gas_parameter = borrow_global_mut<GasParameter>(@aptos_framework);
+    public(friend) fun current_items_and_bytes(): (u64, u64) acquires StateStorageUsage {
+        assert!(
+            exists<StateStorageUsage>(@aptos_framework),
+            error::not_found(ESTATE_STORAGE_USAGE)
+        );
         let usage = borrow_global<StateStorageUsage>(@aptos_framework);
-        gas_parameter.usage = usage.usage;
+        (usage.usage.items, usage.usage.bytes)
     }
 
     /// Warning: the result returned is based on the base state view held by the
     /// VM for the entire block or chunk of transactions, it's only deterministic
     /// if called from the first transaction of the block because the execution layer
     /// guarantees a fresh state view then.
     native fun get_state_storage_usage_only_at_epoch_beginning(): Usage;
+
+    #[test_only]
+    public fun set_for_test(epoch: u64, items: u64, bytes: u64) acquires StateStorageUsage {
+        assert!(
+            exists<StateStorageUsage>(@aptos_framework),
+            error::not_found(ESTATE_STORAGE_USAGE)
+        );
+        let usage = borrow_global_mut<StateStorageUsage>(@aptos_framework);
+        usage.epoch = epoch;
+        usage.usage = Usage {
+            items,
+            bytes
+        };
+    }
+
+    // ======================== deprecated ============================
+    friend aptos_framework::reconfiguration;
```

Comment:
Can we just remove this then?

Reply:
it breaks the module_upgrade_test

Reply:
Weird cc @wrwg. I'm not aware of friend statements being checked in compatibility validation

Reply:
The friend_a_private feature is merged, but the feature isn't on by default. And it can't be turned on before rolled out, strictly spoken (because of replay).

For now any changes to friend function or module declarations will cause the upgrade to fail. But we can clean this once we upgraded testnet. So keep everything for now, remove later.

Reply:
I think it changes visibility. It took me a while to figure it out.

Reply:
The `friend M` decls need to be a superset of the old module. With the new feature this check is not longer happening, but you can't use that yet.